src/main/java/com/exasol/adapter/dynamodb/mapping/AbstractTableMappingFactory.java

```diff
@@ -41,73 +25,140 @@ protected AbstractTableMappingFactory() {
+                final DocumentPathExpression.Builder sourcePath) {
+            final ColumnMappingDefinitionKeyTypeReader.KeyType columnsKeyType = new ColumnMappingDefinitionKeyTypeReader()
+                    .readKeyType(definition);
+            if (columnsKeyType.equals(ColumnMappingDefinitionKeyTypeReader.KeyType.NO_KEY)) {
+                this.nonKeyColumns.add(column);
+            } else {
+                if (this.keyType != columnsKeyType
+                        && this.keyType != ColumnMappingDefinitionKeyTypeReader.KeyType.NO_KEY) {
+                    throw new ExasolDocumentMappingLanguageException(sourcePath.build().toString()
+                            + ": This table already has a key of different type (global/local).  "
+                            + "Please either define all keys of the table local or global.");
+                }
+                this.keyType = columnsKeyType;
+                this.keyColumns.add(column);
+            }
+        }
+
+        private void queueAddingNestedTable(final JsonObject definition,
+                final DocumentPathExpression.Builder sourcePath, final String propertyName) {
+            this.hasNestedTable = true;
+            this.nestedTableReaderQueue
+                    .add(parentTable -> new NestedTableMappingFactory(parentTable, propertyName, sourcePath)
+                            .readMappingDefinition(definition));
+        }
+
+        private String getMappingType(final JsonObject definition, final DocumentPathExpression.Builder sourcePath) {
+            final Set<String> keys = definition.keySet();
+            if (keys.isEmpty()) {
+                return "";
+            } else if (keys.size() == 1) {
+                return keys.iterator().next();
+            } else {
+                throw new ExasolDocumentMappingLanguageException(
+                        sourcePath.build().toString() + ": Please define only one mapping for one property.");
+            }
+        }
+
+        private void visitChildren(final JsonObject definition, final DocumentPathExpression.Builder sourcePath) {
+            for (final String dynamodbPropertyName : definition.keySet()) {
+                final DocumentPathExpression.Builder newBuilder = new DocumentPathExpression.Builder(sourcePath)
+                        .addObjectLookup(dynamodbPropertyName);
+                visitMapping(definition.getJsonObject(dynamodbPropertyName), newBuilder, dynamodbPropertyName, false);
+            }
+        }
+
+        public List<ColumnMappingDefinition> getAllColumns() {
+            final List<ColumnMappingDefinition> union = new ArrayList<>();
+            union.addAll(this.keyColumns);
+            union.addAll(this.nonKeyColumns);
+            return union;
+        }
+
+        public List<NestedTableReader> getNestedTableReaderQueue() {
+            return this.nestedTableReaderQueue;
+        }
+
+        public boolean isHasNestedTable() {
```

Comment:
```suggestion
        public boolean isNestedTable() {
```

Reply:
You also have abstract method above with same name, remove it?

Reply:
These are two different methods. This one is in the nested class and should be `hasNestedTable()` the other one is in the outer class and is called `isNestedTable()`